components/test_raftstore/src/pd.rs

```diff
@@ -817,8 +817,11 @@ impl TestPdClient {
                 .wait()
                 .unwrap()
                 .unwrap();
-            if now.get_start_key() != region.get_start_key()
-                || now.get_end_key() != region.get_end_key()
+            if (now.get_start_key() != region.get_start_key()
+                && self.get_region(region.get_start_key()).is_ok())
+                || (now.get_end_key() != region.get_end_key()
+                    && !now.get_end_key().is_empty()
```

Comment:
what I mean here is adding an assertion like `assert(!now.get_end_key().is_empty())` to ensure the new end key must be not-empty